pyg_lib/csrc/sampler/cpu/neighbor_kernel.cpp

```diff
@@ -223,11 +223,22 @@ class NeighborSampler {
 
     // Case 2: Multinomial sampling:
     else {
-      const auto index = at::multinomial(weight, count, replace);
-      const auto index_data = index.data_ptr<int64_t>();
-      for (size_t i = 0; i < index.numel(); ++i) {
-        add(row_start + index_data[i], global_src_node, local_src_node,
-            dst_mapper, out_global_dst_nodes);
+      if (replace) {
+        const auto index = at::multinomial(weight, count, replace);
+        const auto index_data = index.data_ptr<int64_t>();
+        for (size_t i = 0; i < index.numel(); ++i) {
+          add(row_start + index_data[i], global_src_node, local_src_node,
+              dst_mapper, out_global_dst_nodes);
+        }
```

Comment:
I think we can unify this logic outside the `if`/`else` to avoid the duplicated code here.